pkg/migrations/sql/idempotent/metric-metadata.sql

```diff
@@ -0,0 +1,58 @@
+CREATE OR REPLACE FUNCTION prom_api.insert_metric_metadata(t TIMESTAMPTZ, metric_family_name TEXT, metric_type TEXT, metric_unit TEXT, metric_help TEXT)
+RETURNS BIGINT
+AS
+$$
+    DECLARE
+        num_rows BIGINT;
+    BEGIN
+        INSERT INTO _prom_catalog.metadata (last_seen, metric_family, type, unit, help)  VALUES (t, metric_family_name, metric_type, metric_unit, metric_help)
+            ON CONFLICT (metric_family, type, unit, help) DO
+                UPDATE SET last_seen = t;
+        GET DIAGNOSTICS num_rows = ROW_COUNT;
+        RETURN num_rows;
+    END;
+$$ LANGUAGE plpgsql;
+GRANT EXECUTE ON FUNCTION SCHEMA_PROM.insert_metric_metadata(TIMESTAMPTZ, TEXT, TEXT, TEXT, TEXT) TO prom_writer;
+
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.get_metric_metadata(metric_family_name TEXT)
+RETURNS TABLE (metric_family TEXT, type TEXT, unit TEXT, help TEXT)
+AS
+$$
+    SELECT metric_family, type, unit, help FROM SCHEMA_CATALOG.metadata WHERE metric_family = metric_family_name ORDER BY last_seen LIMIT 1
+$$ LANGUAGE SQL;
+GRANT EXECUTE ON FUNCTION SCHEMA_PROM.get_metric_metadata(TEXT) TO prom_writer;
+
+-- metric_families should have unique elements, otherwise there will be duplicate rows in the returned table.
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.get_multiple_metric_metadata(metric_families TEXT[])
```

Comment:
is this ever used?

Reply:
No, this is for users to write easier querying. I will document these in the docs PR.